pkg/syncer/resource_syncer_test.go

```diff
@@ -1183,6 +1185,37 @@ func testRequeueResource() {
 	})
 }
 
+func testTrimResourceFields() {
+	d := newTestDriver(test.LocalNamespace, "", syncer.LocalToRemote)
+
+	BeforeEach(func() {
+		d.config.ResyncPeriod = time.Millisecond * 100
+
+		d.resource.SetManagedFields([]metav1.ManagedFieldsEntry{
+			{
+				Manager:    "kubectl",
+				Operation:  metav1.ManagedFieldsOperationApply,
+				APIVersion: "v1",
+				Time:       ptr.To(metav1.Now()),
+				FieldsType: "FieldsV1",
+				FieldsV1:   ptr.To(metav1.FieldsV1{}),
+			},
+		})
+
+		d.addInitialResource(d.resource)
+	})
+
+	It("should remove ManagedFields from created resources", func() {
+		obj, exists, err := d.syncer.GetResource(d.resource.Name, d.resource.Namespace)
+		Expect(err).To(Succeed())
+		Expect(exists).To(BeTrue())
+		Expect(resourceutils.MustToMeta(obj).GetManagedFields()).Should(BeNil())
+
+		// Sleep a little so a re-sync occurs and doesn't cause a data race.
+		time.Sleep(200)
```

Comment:
Shouldn’t something be checked after the sleep?

Reply:
There's nothing we can check here but Ginkgo will fail with a data race if it occurs. So we're just trying to trigger a data race here.